js/test/test.js

```diff
@@ -344,82 +419,125 @@ async function testExchange (exchange) {
+        'fetchTradingFees': [exchange],
+        'fetchStatus': [exchange],
+        'fetchOrders': [exchange, symbol],
+        'fetchOpenOrders': [exchange, symbol],
+        'fetchClosedOrders': [exchange, symbol],
+        'fetchMyTrades': [exchange, symbol],
+        'fetchLeverageTiers': [exchange, symbol],
+        'fetchOpenInterestHistory': [exchange, symbol],
+        'fetchPositions': [exchange, symbol],
+        'fetchLedger': [exchange, code],
+        'fetchTransactions': [exchange, code],
+        'fetchDeposits': [exchange, code],
+        'fetchWithdrawals': [exchange, code],
+        'fetchBorrowRates': [exchange, code],
+        'fetchBorrowRate': [exchange, code],
+        'fetchBorrowInterest': [exchange, code],
+        'fetchBorrowInterest': [exchange, code, symbol],
+        'addMargin': [exchange, symbol],
+        'reduceMargin': [exchange, symbol],
+        'setMargin': [exchange, symbol],
+        'setMarginMode': [exchange, symbol],
+        'setPositionMode': [exchange, symbol],
+        'setLeverage': [exchange, symbol],
+        'cancelAllOrders': [exchange, symbol],
+        'cancelOrder': [exchange, symbol],
+        'cancelOrders': [exchange, symbol],
+        'fetchCanceledOrders': [exchange, symbol],
+        'fetchClosedOrder': [exchange, symbol],
+        'fetchOpenOrder': [exchange, symbol],
+        'fetchOrder': [exchange, symbol],
+        'fetchOrderTrades': [exchange, symbol],
+        'fetchPosition': [exchange, symbol],
+        'fetchFundingHistory': [exchange, symbol],
+        'fetchDeposit': [exchange, code],
+        'createDepositAddress': [exchange, code],
+        'fetchDepositAddress': [exchange, code],
+        'fetchDepositAddresses': [exchange, code],
+        'fetchDepositAddressesByNetwork': [exchange, code],
+        'editOrder': [exchange, symbol],
+        'fetchBorrowRateHistory': [exchange, symbol],
+        'fetchBorrowRatesPerSymbol': [exchange, symbol],
+        'fetchLedgerEntry': [exchange, code],
+        'fetchPositionsRisk': [exchange, symbol],
+        'fetchWithdrawal': [exchange, code],
+        'transfer': [exchange, code],
+        'withdraw': [exchange, code],
+    };
+
+    const testNames = Object.keys (tests);
+    const errors = [];
+    const promises = [];
+    for (let i = 0; i < testNames.length; i++) {
+        const testName = testNames[i];
+        const testArgs = tests[testName];
+        promises.push (testSafePrivate (testName, ...testArgs));
     }
 
-    await test ('fetchTransactions', exchange, code);
-    await test ('fetchDeposits', exchange, code);
-    await test ('fetchWithdrawals', exchange, code);
-    await test ('fetchBorrowRate', exchange, code);
-    await test ('fetchBorrowRates', exchange);
-    await test ('fetchBorrowInterest', exchange, code);
-    await test ('fetchBorrowInterest', exchange, code, symbol);
-
-    if (exchange.extendedTest) {
+    const results = await Promise.all (promises);
```

Comment:
@ttodua not sure about this, we might want to know the orders, let's say in the future we create an order and then we fetch it, it's important to run `createOrder` first

Reply:
why? createOrder tests are separate file, isolated, where it has it's own fetch inside it. so here , the point of testing in bulk is that we dont care about order (only `signIn` as an exception). 
so, promise.all seems better to me. but as you'll say, will do so.

Reply:
@ttodua you might need to create an order first to have `fetchOrders/fetchClosedOrders/etc` returning anything. Nonetheless yes we can make some tests first before deciding